src/schedule/zephyr_dp_schedule.c

```diff
@@ -345,7 +345,7 @@ int scheduler_dp_task_init(struct task **task,
 	/* success, fill the structures */
 	task_memory->task.priv_data = &task_memory->pdata;
 	task_memory->pdata.thread_id = thread_id;
-	task_memory->pdata.p_stack = p_stack;
+	task_memory->pdata.p_stack = z_soc_cached_ptr(p_stack);
```

Comment:
here too, alloc will return cached pointer. no need to change
to be 100% accurate - a flag SOF_MEM_CAPS_CACHE indeed may be added to allocation
It won't change a bit in flow, just be more clear to read

task_memory = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM **| SOF_MEM_CAPS_CACHE**, sizeof(*task_memory));


p_stack = rballoc_align(0, SOF_MEM_CAPS_RAM **| SOF_MEM_CAPS_CACHE**, stack_size, Z_KERNEL_STACK_OBJ_ALIGN);

Reply:
Use this for `rballoc()` https://github.com/thesofproject/sof/blob/1319163ea2c6d8bea6fd7189dea6275ac5e34bf5/src/audio/module_adapter/module_adapter.c#L326-L327 and add `__sparse_cache` to `p_stack` definition